### PR TITLE
fix: unused enum variants

### PR DESCRIPTION
**Pull request by pront**
Failure: https://github.com/vectordotdev/vector/actions/runs/7108315012/attempts/2

Failure was introduced here: https://github.com/vectordotdev/vector/pull/18929/files

### CODE REVIEW

File: src/sinks/datadog/test_utils.rs [hunk 1]
```diff
@@ -0,0 +1,39 @@
+#![cfg(test)]
```
Comment (pront) [inline src/sinks/datadog/test_utils.rs:1]: Reminder to remove this, but don't want to interupt the current run

**Review by dsmith3197 — APPROVED** [review]

**Pull request merged**